mail_tracking/static/src/js/mail_tracking.js

```diff
@@ -1,102 +1,105 @@
-            _.each(trackings, function (tracking, id) {
-                msgs[id].partner_trackings = tracking;
+    ChatThread.include({
+        on_tracking_partner_click: function (event) {
+            var partner_id = this.$el.find(event.currentTarget).data('partner');
+            var state = {
+                'model': 'res.partner',
+                'id': partner_id,
+                'title': _t("Tracking partner"),
+            };
+            event.preventDefault();
+            this.action_manager.do_push_state(state);
+            var action = {
+                type:'ir.actions.act_window',
+                view_type: 'form',
+                view_mode: 'form',
+                res_model: 'res.partner',
+                views: [[false, 'form']],
+                target: 'current',
+                res_id: partner_id,
+            };
+            this.do_action(action);
+        },
+        on_tracking_status_click: function (event) {
+            var tracking_email_id = $(event.currentTarget).data('tracking');
+            var state = {
+                'model': 'mail.tracking.email',
+                'id': tracking_email_id,
+                'title': _t("Message tracking"),
+            };
+            event.preventDefault();
+            this.action_manager.do_push_state(state);
+            var action = {
+                type:'ir.actions.act_window',
+                view_type: 'form',
+                view_mode: 'form',
+                res_model: 'mail.tracking.email',
+                views: [[false, 'form']],
+                target: 'new',
+                res_id: tracking_email_id,
+            };
+            this.do_action(action);
+        },
+        bind_events: function () {
+            this.$el.on('click', '.o_mail_action_tracking_partner',
+                        this.on_tracking_partner_click);
+            this.$el.on('click', '.o_mail_action_tracking_status',
+                        this.on_tracking_status_click);
+        },
+        init: function (parent, options) {
+            this._super.apply(this, arguments);
+            this.action_manager = this.findAncestor(function(ancestor){
+                return ancestor instanceof ActionManager;
             });
-            render_super.apply(self, [messages, options]);
-        });
-    },
-
+        },
+        start: function () {
+            this._super();
+            this.bind_events();
+        },
+        render: function(messages, options) {
+            var self = this, render_super = this._super,
+                msgs = {},
+                msg_ids = [];
+            // Update trackings (async) each time we re-render thread
+            _.each(messages, function (message) {
+                msgs[message.id] = message;
+                msg_ids.push(message.id);
+            });
+            self._rpc({
+                model: 'mail.message',
+                method: 'tracking_status',
```

Comment:
This thing you're doing here is not exactly wrong, but think about it: every time the chat thread is rendered (which can happens i.e. when removing a message), you're making an additional RPC call and waiting for it to resolve until you actually render the thread.

Since reading mail threads is so common in Odoo, they designed it in a way that uses the web bus and caches messages internally, so it reduces RPC calls and improves responsiveness. Here, we're introducing a bottleneck that honestly I'd prefer to avoid.

By digging into the made calls, you can see that [the web client ultimately calls the model's `message_fetch` method](https://github.com/odoo/odoo/blob/71087021f7a7881c023ac46ea9d0391be966a569/addons/mail/static/src/js/chat_manager.js#L867). Then, **it would be much better to simply overwrite that method in the backend** and avoid these RPC calls. The JS code would be much less too now.

This would be a really big performance improvement. Would you do it?

Reply:
Well, I'd say that what you propose is alredy happening here: https://github.com/OCA/social/pull/244/files#diff-4e1cf268ea32e1266c1e39d18db0d429R66 Wich is a common point in the already exisiting rpc calls. Indeed, this code could be erased with no harm. Don't you think this method was made to have tighter check of the thread messages status?

Reply:
I understand. For what I can see, this could be erased. Could you check please?